lib/core/src/Cardano/Wallet/Primitive/Types/Coin.hs

```diff
@@ -14,6 +14,8 @@ module Cardano.Wallet.Primitive.Types.Coin
     ( -- * Type
       Coin (..)
     , coinQuantity
+    , coinToInteger
+    , coinToNatural
```

Comment:
Looks like coinToNatural isn't used in this PR?
coinToInteger is used for difference - could it be replaced with a difference function?

Reply:
Yes indeed, it used to be used before your PR which changed many of the `Quantity` to `Coin`. The `coinToInteger` is only used in the readWithdrawal function; Integer are used here because the difference may actually underflow / be negative. I'll see if that can be written differently using only `distance` or `subtractCoin` :+1: